common/djangoapps/course_modes/views.py

```diff
@@ -135,6 +136,7 @@ def get(self, request, course_id, error=None):
 
         return render_to_response("course_modes/choose.html", context)
 
+    @method_decorator(transaction.non_atomic_requests)
```

Comment:
I don't understand why both `commit_on_success_with_read_committed` and `transaction.non_atomic_requests`. Seems to have the effect of setting the transaction isolation only?